compiler/src/java_plugin/cpp/java_generator.cpp

```diff
@@ -143,10 +143,23 @@ static std::set<std::string> java_keywords = {
   "false",
 };
 
+// Methods on java.lang.Object that take no arguments.
+static std::set<std::string> java_object_methods = {
+    "clone",
+    "finalize",
+    "getClass",
+    "hashCode",
+    "notify",
+    "notifyAll",
+    "toString",
+    "wait",
+};
+
 // Adjust a method name prefix identifier to follow the JavaBean spec:
 //   - decapitalize the first letter
 //   - remove embedded underscores & capitalize the following letter
-//  Finally, if the result is a reserved java keyword, append an underscore.
+//  Finally, if the result is a reserved java keyword or an Object method,
```

Comment:
This will change the existing behavior that is working for all the other cases. We should only change the blocking stub, as it is the only one that has zero-arg methods.